### PR TITLE
pull request of requested feature implementation 

### PR DESCRIPTION
**Pull request by sivashanmugam-funtoot**

### CODE REVIEW

**Comment by sivashanmugam-funtoot** [comment]
merging pull request

File: renderer/js/mtftemplate.js [hunk 1]
```diff
@@ -1,42 +1,209 @@
...
+}
+
+/**
+ * image will be shown in popup
+ * @memberof org.ekstep.questionunit.mtf.mtftemplate
+ */
+MTFController.showImageModel = function (event, imageSrcType) {
+  var eventData;
+  if (imageSrcType == "background-image") {
+    eventData = event.target.style.backgroundImage.slice(4, -1).replace(/"/g, "");
```
Comment (vinukumar-vs) [inline renderer/js/mtftemplate.js]: relaying on style background image path is not good. Redesign/rethink of this workflow.

File: renderer/js/mtftemplate.js [hunk 2]
```diff
@@ -1,42 +1,209 @@
...
+  })
+  $(MTFController.constant.qsMTFElement).append(templateData);
+},
+
+/**
+ * onclick overlay or X button the popup will be hide
+ * @memberof org.ekstep.questionunit.mtf.mtftemplate
+ */
+MTFController.hideImageModel = function () {
+  $("#image-model-popup").remove();
```
Comment (vinukumar-vs) [inline renderer/js/mtftemplate.js]: Handle through event parameter instead of id.
MTFController.hideImageModel(event)

File: renderer/plugin.js [hunk 3]
```diff
@@ -12,15 +12,29 @@ org.ekstep.questionunitmtf.RendererPlugin = org.ekstep.contentrenderer.questionU
...
+  },
   setQuestionTemplate: function () {
-    this._question.template = QS_MTFTemplate.htmlLayout;
+    MTFController.initTemplate(this);// eslint-disable-line no-undef
   },
   preQuestionShow: function (event) {
     this._super(event);
+    var templateTypeContent;
+    if (this._question.config.layout == this._constant.vertial) { // eslint-disable-line no-undef
+      templateTypeContent = MTFController.getVerticalLayout;
```
Comment (vinukumar-vs) [inline renderer/plugin.js]: MTFController.getVerticalLayout is a funciton not a variable.

File: renderer/plugin.js [hunk 4]
```diff
@@ -12,15 +12,29 @@ org.ekstep.questionunitmtf.RendererPlugin = org.ekstep.contentrenderer.questionU
...
-    this._question.template = QS_MTFTemplate.htmlLayout;
+    MTFController.initTemplate(this);// eslint-disable-line no-undef
   },
   preQuestionShow: function (event) {
     this._super(event);
+    var templateTypeContent;
+    if (this._question.config.layout == this._constant.vertial) { // eslint-disable-line no-undef
+      templateTypeContent = MTFController.getVerticalLayout;      
+    }else{
+      templateTypeContent = MTFController.getHorizontalLayout;  //default is horizontal
```
Comment (vinukumar-vs) [inline renderer/plugin.js]: MTFController.getHorizontalLayout is a funciton not a variable.

File: renderer/plugin.js [hunk 5]
```diff
@@ -31,21 +45,33 @@ org.ekstep.questionunitmtf.RendererPlugin = org.ekstep.contentrenderer.questionU
...
     if (this._question.state && this._question.state.val) {
       this._selectedAnswers = this._question.state.val.lhs;
-      QS_MTFTemplate.selAns = this._question.state.val.lhs;
+      MTFController.selAns = this._question.state.val.lhs;
       this._question.data.option.optionsRHS = this._question.state.val.rhs;
     }
     QSTelemetryLogger.logEvent(QSTelemetryLogger.EVENT_TYPES.ASSESS); // eslint-disable-line no-undef
+
+    //here have to change based on the type of the question whether horizontal or vertial length (currently horizontal)
+    if (lhsOptions.length == 3) {
```
Comment (vinukumar-vs) [inline renderer/plugin.js]: what is the default use-case here. If lhsOptions >3 or <5?

File: renderer/plugin.js [hunk 6]
```diff
@@ -31,21 +45,33 @@ org.ekstep.questionunitmtf.RendererPlugin = org.ekstep.contentrenderer.questionU
...
+      MTFController.optionsWidth = 'width33';
+    } else if (lhsOptions.length == 4) {
+      MTFController.optionsWidth = 'width25';
+    } else if (lhsOptions.length == 5) {
+      MTFController.optionsWidth = 'width20';
+    }
+
+    //this height[NUMBER]option is a class to apply for vertical mtf template based on number of option
     if (lhsOptions.length == 3) {
-      QS_MTFTemplate.optionsWidth = 'width33';
```
Comment (vinukumar-vs) [inline renderer/plugin.js]: what is the default use-case here. If optionsWidth >3 or <5?

File: renderer/styles/style.css [hunk 7]
```diff
@@ -185,4 +210,138 @@ header.mtf-header {
...
+    color: white;
+    bottom: 87%;
+}
+
+img.qc-question-fullimage{
+    width:83%;
+    height:100%;
+}
+
+.popup{
```
Comment (vinukumar-vs) [inline renderer/styles/style.css]: Dont use .popup class to override the default hegiht to 90%. Use the specific class name to override.

File: renderer/js/mtftemplate.js (hunk 1 above)
Comment (ramj-funtoot) [reply renderer/js/mtftemplate.js]: @vinukumar-vs This was already existing, so left this as is. Yes, you are correct, we should not rely on the background image. Will work on a better flow. @sivashanmugam-funtoot

File: editor/controllers/horizontal_controller.js [hunk 8]
```diff
@@ -175,96 +175,113 @@ angular.module('mtfApp', [])
...
-          "ver": data.target.ver,
-          "type": data.target.type
-        },
-        "plugin": {
-          "id": "org.ekstep.questionunit.mtf",
-          "ver": "1.0"
-        }
-      })
+      var plugin = {
+        "id": "org.ekstep.questionunit.mtf",
```
[the remark this answers is not in the document]
Comment (ramj-funtoot) [reply editor/controllers/horizontal_controller.js]: Good point. Fixing it.

File: editor/controllers/horizontal_controller.js [hunk 9]
```diff
@@ -175,96 +175,113 @@ angular.module('mtfApp', [])
...
-            $scope.mtfFormData.option.optionsLHS[id].audio = org.ekstep.contenteditor.mediaManager.getMediaOriginURL(data.assetMedia.src);
-            $scope.optionsMedia.audio[id] = tempAudio;
-          } else if (type == 'RHS') {
-            $scope.mtfFormData.option.optionsRHS[id].audio = org.ekstep.contenteditor.mediaManager.getMediaOriginURL(data.assetMedia.src);
-            $scope.optionsMedia.audio[id] = tempAudio;
-          }
+        search_filter: {} // All composite keys except mediaType
+      }
+      //Defining the callback function of mediaObject before invoking asset browser
+      mediaObject.callback = function(data) {
```
[the remark this answers is not in the document]
Comment (ramj-funtoot) [reply editor/controllers/horizontal_controller.js]: Fixing these as well.

File: editor/controllers/horizontal_controller.js [hunk 10]
```diff
@@ -175,96 +175,113 @@ angular.module('mtfApp', [])
...
-          "id": "org.ekstep.questionunit.mtf",
-          "ver": "1.0"
-        }
-      })
+      var plugin = {
+        "id": "org.ekstep.questionunit.mtf",
+        "ver": "1.0"
+      }
+      data.form = 'question-creation-mtf-form';
+      $questionServices.generateTelemetry(data);
```
[the remark this answers is not in the document]
Comment (ramj-funtoot) [reply editor/controllers/horizontal_controller.js]: Yes, fixing this as well.

File: editor/controllers/horizontal_controller.js [hunk 11]
```diff
@@ -175,96 +175,113 @@ angular.module('mtfApp', [])
...
+        telemetryObject.target.id = 'questionunit-mtf-rhs-delete-imgae';
         $scope.mtfFormData.option.optionsRHS[id].image = '';
         delete $scope.optionsMedia.image[id];
       }
+      $scope.generateTelemetry(telemetryObject)
     }
     $scope.deleteAudio = function(id, type) {
+      var telemetryObject = {type: 'TOUCH', id: 'button', target: {id: '', ver: '', type: 'button'}};
       if (id == 'q') {
+        telemetryObject.target.id = 'questionunit-mtf-delete-audio';
```
[the remark this answers is not in the document]
Comment (ramj-funtoot) [reply editor/controllers/horizontal_controller.js]: This code is also refactored now to for code reuse. Will fix the strings as you have suggested.

File: renderer/plugin.js (hunk 3 above)
Comment (ramj-funtoot) [reply renderer/plugin.js]: Yes Vinu. `MTFController.getVerticalLayout` is a function. This function is being called below based on the selected layout.

File: renderer/plugin.js (hunk 4 above)
Comment (ramj-funtoot) [reply renderer/plugin.js]: Yes Vinu. `MTFController.getHorizontalLayout` is a function. This function is being called below based on the selected layout.

**Pull request merged**